# <u> Mini Project 1 : Project Option 2: Student Marks Analyzer</u>

-By Raunak Kumar Singh

#**<u>Problem Statement</u>**

We have to make a program which inputs the student marks and returns us:-

*   Total Marks
*   Average Marks
*   Highest Marks
*   Lowest Marks
*   Assigns Grade to students based on their marks

The Program should acheive this by storing the marks in a NumPy Array.

#**<u>Logic Explaination</u>**

I first import numpy because I want to use numpy arrays and its functions in order to calculate easily. I then input the marks in a list via a loop. I then use if-elif-else conditional block to give the grade of the student. After that, I make a NumPy array from that list. I use sum(),mean(),max(),min() to calculate the sum,mean,max marks, min marks. I also use round() to round the mean to 2 decimal digits.

In [ ]:
#Project Option 2: Student Marks Analyzer

import numpy

lst=[]
x=int(input("Enter the no. of students:- "))

for i in range(x):
    lst.append(int(input(f"Enter the marks of {i+1}th student:- ")))

for i in range(x):
    if(lst[i]>=85):
        print(f"The Grade of Student {i+1} is A.")
    elif(lst[i]>=65):
        print(f"The Grade of Student {i+1} is B.")
    elif(lst[i]>=50):
        print(f"The Grade of Student {i+1} is C.")
    else:
        print(f"Student {i+1} has Failed.")

arr=numpy.array(lst)

print(f"Total Marks are {numpy.sum(arr)}")
print(f"Average Marks are {round(numpy.mean(arr),2)}")
print(f"Highest Marks are {numpy.max(arr)}")
print(f"Lowest Marks are {numpy.min(arr)}")

Enter the no. of students:- 4
Enter the marks of 1th student:- 55
Enter the marks of 2th student:- 34
Enter the marks of 3th student:- 98
Enter the marks of 4th student:- 77
The Grade of Student 1 is C.
Student 2 has Failed.
The Grade of Student 3 is A.
The Grade of Student 4 is B.
Total Marks are 264
Average Marks are 66.0
Highest Marks are 98
Lowest Marks are 34


#**<u>Sample Output</u>**

Enter the no. of students:- 4

Enter the marks of 1th student:- 55

Enter the marks of 2th student:- 34

Enter the marks of 3th student:- 98

Enter the marks of 4th student:- 77

The Grade of Student 1 is C.

Student 2 has Failed.

The Grade of Student 3 is A.

The Grade of Student 4 is B.

Total Marks are 264

Average Marks are 66.0

Highest Marks are 98

Lowest Marks are 34
